OKTA-537325 OIN API service integration (#3708)
okta/okta-developer-docs: packages/@okta/vuepress-site/docs/guides/oin-api-service-overview/index.md

---
title: API service integrations in the OIN
meta:
  - name: description
    content: Provides an overview of why and when to build API service integrations for the Okta Integration Network.
---

<ApiLifecycle access="ea" />

> **Note:** The API Service Integration is a [Self-service Early Access (EA)](/docs/concepts/feature-lifecycle-management/#self-service-features) feature. Enable this feature in your org to obtain the API service integration capability.
<!-- Need OAUTH_ADMIN_CONSENT_DIALOG FF enabled in monolith org -->

> Service apps that access or modify Okta resources, like system logs, apps, sessions, and policies can be published in the Okta Integration Network (OIN) as an API service integration. This integration type allows your service app to access your customer tenant Okta org through [Okta management APIs](/docs/reference/core-okta-api/) using OAuth 2.0.

Author: ```suggestion
You can publish any service app that accesses or modifies Okta resources (such as system logs, apps, sessions, or policies) in the Okta Integration Network (OIN) as an API service integration. This integration type allows your service app to access your customer tenant Okta org through [Okta management APIs](/docs/reference/core-okta-api/) using OAuth 2.0.
```
the list in parenthesis are examples, not an exhaustive list

> Previously, if you have a service app requiring Okta management API access, you need to use an API token that was tied to a specific Okta user and you can't restrict access to certain resources. With an OAuth 2.0 API service integration, your service app can have secure, reliable, and granular access to Okta APIs without being associated with a user.

Author: I like your suggestion! I'm tweaking it a bit for style and removing a sentence.
```suggestion
Before this feature, a service requiring Okta management API access used an API token tied to a specific Okta user. This approach meant you couldn't restrict the service’s access to particular resources. However, with an OAuth 2.0 API service integration, your service app can have secure, reliable, and granular access to Okta APIs without being associated with a user.
```

> When your service app integration is in the Okta Integration Network (OIN), your customers can discover and configure your integration through the OIN catalog in their own Okta tenant org. Configuration is easy and consistent for your customers because you've already done the hard work by building the integration with the required configuration instructions. Customers trust that integrations in the OIN are secure and reliable because it's Okta verified.

Author: Thanks for the suggestion! I have to inject the "integration" word in the first sentence because the app is not in the catalog, it's the integration.
```suggestion
When your service app integration is listed in the Okta Integration Network (OIN) catalog, your customers can discover and configure it in their own Okta tenant org. Configuration is easy and consistent for your customers because you've already done the hard work by building the integration with the required configuration instructions. Customers trust that integrations in the OIN are secure and reliable because they're Okta-verified.
```

## When to build an API service integration

API service integrations access [Okta management APIs](/docs/reference/core-okta-api/) using OAuth 2.0 and are sometimes referred to as service-to-service or machine-to-machine integrations. This API access represents the application itself rather than a particular user. Consider the following scenarios to determine if you need to build a service-based or a user-based OAuth 2.0 API integration.

| &nbsp; |  Service-based API access | User-based API access |
| ------ | :------------------- | :----------------------- |

> | **Use cases** | <ul><li>Log entries need to have the application name instead of a user name for auditing purposes.</li> <li>API requests aren't initiated by a user action (for example, a user clicking a button or starting a script).</li> <li>The application that requests API access is hosted on a secure server (this type of application is also known as a confidential client).</li></ul> | <ul><li>A new custom application is configured from a user entry form.</li> <li>An admin user takes action by making API requests.</li></ul> |

Author: These examples are from the PM. Due to the release schedule, I would like to keep the terms used for this iteration.

| **Examples** | <ul><li>An app that access certain types of Okta system log entries for occasional analysis</li> <li>An app that sends risk signals to Okta</li> <li>An app that occasionally syncs Okta user profile information to another system</li></ul> |  <ul><li>An IT support ticketing system integration that allows an Okta admin to assign an application directly from the access request support ticket</li></ul> |
| **Best-fit API integration** | If these use cases and examples describe your needs, then implementing an API service integration for the OIN is a best-fit solution. See [Build an API service integration](/docs/guides/build-api-integration/).  | If these use cases and examples describe your needs, then implementing a user-based API integration is a better solution. User-based API integrations aren't currently supported in the OIN. See [Implement OAuth for Okta](/docs/guides/implement-oauth-for-okta/main/) to learn more about user-based API integrations. |

> ## API service integration benefits

Author: Thanks for the suggestion! I'll implement this in a local commit.

Before API service integrations were supported in the OIN, Okta API tokens were a common way for integrations to access the Okta management APIs. API service integrations offer improved security and reliability:

> | &nbsp; | &nbsp; |

Author: I think it's for the thicker top line for the table?

| &nbsp; |  Service-based API access | User-based API access |
| ------ | :------------------- | :----------------------- |

> | **Use cases** | <ul><li>For auditing purposes, log entries need to to be associated with an application name instead of a username.</li> <li>API requests aren't initiated by a user action (for example, a user clicking a button or starting a script).</li> <li>The application that requests API access is hosted on a secure server (this type of application is also known as a confidential client).</li></ul> | <ul><li>A new custom application is configured from a user entry form.</li> <li>An admin user takes action by making API requests.</li></ul> |

Reviewer: ```suggestion
| **Use cases** | <ul><li>For auditing purposes, you need to associate log entries with an application name instead of a username.</li> <li>API requests aren't initiated by a user action (for example, a user clicking a button or starting a script).</li> <li>The application that requests API access is hosted on a secure server (this type of application is also known as a confidential client).</li></ul> | <ul><li>A new custom application is configured from a user entry form.</li> <li>An admin user takes action by making API requests.</li></ul> |
```

| **Reliability** | Okta API tokens expire after 30 days of inactivity or when a user leaves the Okta org. After being authorized, API service integrations can request a new access token whenever necessary and without manual action from an admin. |

> | **Rotation** | Okta API tokens must be rotated manually. API service integration access tokens can be rotated programmatically. |

Author: Yes, I believe manually == getting API token from Admin Console, programmatically == using code to monitor when the token expires and fetching a new one when needed

> | **Rotation** | Okta API tokens must be rotated manually. API service integration access tokens can be rotated programmatically. |

Reviewer 2: @vanngo-okta In that case
```suggestion
| **Rotation** | Okta API tokens must be rotated manually using the Admin Console. API service integration access tokens can be rotated programmatically in the code of your service integration. |
```
